src/main/java/com/example/jjdind49_katibar_zad21_1/ProductController.java

```diff
@@ -0,0 +1,53 @@
+package com.example.jjdind49_katibar_zad21_1;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.Collection;
+
+@Controller
+public class ProductController {
+    private final ProductRepository productRepository;
+
+    public ProductController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping("/")
+    String home() {
+        return "index";
+    }
+
+    @GetMapping("/lista")
+    public String showList(Model model, @RequestParam(required = false, name = "kategoria") Category category) {
+        Collection<Product> products;
+        if (category == null) {
+            products = productRepository.findAll();
+        } else {
+            products = productRepository.findProductByCategory(category);
+            model.addAttribute("category", category);
+        }
+        model.addAttribute("products", products);
+        model.addAttribute("sum", ProductUtility.calculatePricesSum(products));
+        return "lista";
+    }
+
+    @GetMapping("/add")
+    String addForm(Model model) {
+        model.addAttribute("product", new Product());
+        return "add";
+    }
+
+    @PostMapping("/add")
+    String addProductFromForm(Product product) {
+        if (product != null & product.getName() != null & product.getPrice() > 0 & product.getCategory() != null ) {
```

Comment:
Zdania logiczne buduj wykorzystując && (jeżeli jakiś warunek nie będzie prawdziwy, to pozostałe nie będą sprawdzane)
`product.getName()` nigdy nie zwróci null, może być co najwyżej pustym Stringiem, więc zamiast porównania do nulla możesz użyć metody isEmpty() z klasy String